src/Tribe/Events/Attendee_List_Display.php

```diff
@@ -0,0 +1,201 @@
+		} else {
+			$is_showing_attendee_list = $this->is_showing_attendee_list_with_classical_editor( $post );
+		}
+
+		/**
+		 * Returns true if the post is displaying a list of attendees.
+		 *
+		 * You can use this filter to let the system know that you're displaying
+		 * the Attendee List in some other way.
+		 *
+		 * @since TBD
+		 *
+		 * @param bool    $is_showing_attendee_list Whether the post is showing the attendee list or not
+		 * @param WP_Post $post The WP_Post object being checked
+		 * @param bool    $is_using_blocks Whether the post is using Blocks or not
+		 *
+		 * @return bool
+		 */
+		$is_showing_attendee_list = (bool) apply_filters(
+			'tribe_tickets_event_is_showing_attendee_list',
+			$is_showing_attendee_list,
+			$post,
+			$this->is_using_blocks() 
+		);
+
+		update_post_meta( $post->ID, Tribe__Tickets_Plus__Attendees_List::HIDE_META_KEY, $is_showing_attendee_list );
+	}
+
+	/**
+	 * This keeps track of whether the Attendee List is being displayed becase of a shortcode
+	 * in the content, and acts accordingly if said shortcode is removed.
+	 *
+	 * @param WP_Post $post
+	 *
+	 * @return void
+	 */
+	private function track_shortcode_driven_meta( WP_Post $post ) {
+		$is_visible_by_meta                  = Tribe__Tickets_Plus__Attendees_List::is_hidden_on( $post ) === false;
+		$has_attendee_list_shortcode         = has_shortcode( $post->post_content, 'tribe_attendees_list' );
+		$has_attendee_list_by_shortcode_meta = get_post_meta( $post->ID, self::$attendee_list_by_shortcode, true ) === 'yes';
+
+		/*
+		 * If what triggers the Attendee List to display is a shortcode in the content,
+		 * let's save this piece of information so that we revert it when the shortcode is removed.
+		 */
+		if ( ! $is_visible_by_meta && $has_attendee_list_shortcode ) {
+			update_post_meta( $post->ID, self::$attendee_list_by_shortcode, 'yes' );
+		}
+
+		/*
+		 * The shortcode that triggered the Attendee List to display is no longer in the content.
+		 * Thus, we shall update the "Attendee List" meta to "hidden" again.
+		 */
+		if ( $has_attendee_list_by_shortcode_meta && ! $has_attendee_list_shortcode ) {
+			update_post_meta( $post->ID, self::$attendee_list_by_shortcode, 'no' );
+
+			// This updates the main meta to false.
+			add_filter( 'tribe_tickets_event_is_showing_attendee_list', '__return_false' );
+		}
+	}
+
+	/**
+	 * @return bool
+	 */
+	private function is_using_blocks() {
+		$should_load_blocks          = tribe( 'editor' )->should_load_blocks();
+		$is_blocks_active_for_events = tribe_is_truthy( tribe_get_option( 'toggle_blocks_editor', false ) );
+		$is_using_blocks             = $should_load_blocks && $is_blocks_active_for_events;
+
+		/**
+		 * Filter for use in automated tests.
+		 *
+		 * This filter is for internal use only. It is not guaranteed to be maintained,
+		 * and can be removed or changed at any time.
+		 */
+		$is_using_blocks = apply_filters( '__internal_tribe_tickets_is_using_blocks', $is_using_blocks );
```

Comment:
When you add a filter, that's going to be public by definition; I think marking this as internal will just add confusion.
Furthermore, filters are parsed and documented from the doc site auto-parser.
I'd remove the `__internal_` prefix as this is a perfectly legit filter: if you found a need for it, even testing, then it's likely other developers will find that useful too.

Reply:
@lucatume

I understand and agree that this is a useful filter to have. The only problem is that this method `is_using_blocks()` is probably going to be [moved very soon](https://github.com/moderntribe/tribe-common/pull/1334) to Tribe Common. I didn't want to stale this PR while we wait for the more definitive code to be merged.

Perhaps I could just change the filter's tag to `tribe_is_using_blocks` (as if it were already in Common), and then when the PR is approved and ET's Common is updated I personally take care of doing the small refactor of removing this private method and making sure the filter tag remains the same.

What do you think?

Reply:
The second option seems a good one, make sure to remember to do it.

Reply:
Done. Thanks. https://github.com/moderntribe/event-tickets/pull/1633/commits/11138bc05eca923010db9f0a5f22d2cee4527522